File: 1. Easy/589. N-ary Tree Preorder Traversal/RecursiveSolution.java
class Solution {
    public List<Integer> preorder(Node root) {
        List<Integer> res = new ArrayList<>();
        visit(root, res);
        return res;
    }

    private void visit(Node node, List<Integer> tra) {
        if (node == null)
            return;

        tra.add(node.val);
        for (Node c : node.children) {
            visit(c, tra);
        }
    }
}

File: 1. Easy/589. N-ary Tree Preorder Traversal/recursive-solution.js
var preorder = function (root) {
    const list = [];
    const visit = (node, list) => {
        if (node == null)
            return;

        list.push(node.val);
        for (let child of node.children) {
            visit(child, list);
        }
    }

    visit(root, list);
    return list;
};